pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -44,7 +44,24 @@ def addend(request):
 
 
 class TestDatetimeIndexComparisons(object):
-    # TODO: De-duplicate with test_comparisons_nat below
+    @pytest.mark.parametrize('other', [None,
+                                       datetime(2016, 1, 1).date(),
+                                       np.nan])
+    def test_dti_cmp_invalid(self, tz, other):
```

Comment:
hmm, I think leave None and np.nan here is ok (rename this from invalid -> null or something more descriptive).

put the date with a Timestamp / datetime test.